Older versions of randomgen cause `ValueError: numpy.dtype size changed`

**Describe the bug**
I hadn't used my development environment recently and when starting tests I got this error.
```
test/extras/test_csprng.py:6: in <module>
    import randomgen # type: ignore[import-not-found] # noqa F401
.venv/lib/python3.12/site-packages/randomgen/__init__.py:5: in <module>
    from randomgen._register import BitGenerators
.venv/lib/python3.12/site-packages/randomgen/_register.py:3: in <module>
    from randomgen.aes import AESCounter
randomgen/aes.pyx:1: in init randomgen.aes
    ???
randomgen/common.pyx:1: in init randomgen.common
    ???
E   ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

I tried the `pip install` etc. directions, but the older version satisfied dependency requirements, so it didn't upgrade.

**To Reproduce**
```
pip install 'randomgen==1.26.1'
pytest
```

--- a/docs/requirements_notebooks.txt
+++ b/docs/requirements_notebooks.txt
@@ -1,5 +1,5 @@
 matplotlib
-randomgen
+randomgen>=2.0.0
 pandas
 seaborn
 scipy

--- a/python/setup.cfg
+++ b/python/setup.cfg
@@ -40,7 +40,7 @@ opendp =
 # If this changes, update the smoke-test matrix.
 numpy = 
 	numpy
-	randomgen
+	randomgen>=2.0.0
 scikit-learn = 
 	scikit-learn
 	%(numpy)s

